cmake/kconfig.cmake

```diff
@@ -82,7 +86,7 @@ set(
   merge_config_files
   ${BOARD_DEFCONFIG}
   ${CONF_FILE_AS_LIST}
-  ${OVERLAY_CONFIG}
+  ${OVERLAY_FILE_AS_LIST}
```

Comment:
Let's be consistent with how we name the variables.

Usually, we add the suffix _AS_LIST, e.g.

CONF_FILE
CONF_FILE_AS_LIST

so this would be

OVERLAY_CONFIG
OVERLAY_CONFIG_AS_LIST